src/java.net.http/share/classes/jdk/internal/net/http/ConnectionPool.java

```diff
@@ -450,25 +468,22 @@ List<HttpConnection> purgeUntil(Instant now) {
             return closelist;
         }
 
-        // should only be called while holding a synchronization
-        // lock on the ConnectionPool
+        // should only be called while holding the ConnectionPool stateLock.
         java.util.stream.Stream<ExpiryEntry> stream() {
             return list.stream();
         }
 
-        // should only be called while holding a synchronization
-        // lock on the ConnectionPool
+        // should only be called while holding the ConnectionPool stateLock.
```

Comment:
Hello Daniel, all these methods which say a lock needs to be held when they are called, should we add a `assert stateLock.isHeldByCurrentThread();` to make it verifiable? I understand we didn't have a similar assert when the comment said the method needs to be called while holding the monitor, but since we are changing this part now, perhaps we can add those asserts?

Reply:
The `stateLock` variable is not available from within the method:
```
Non-static field 'stateLock' cannot be referenced from a static context
```

Reply:
Thank you, that wasn't clearly noticable in the diff.